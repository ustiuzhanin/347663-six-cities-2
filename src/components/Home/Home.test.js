import React from 'react';
import Home from './Home.jsx';
import ShallowRenderer from 'react-test-renderer/shallow';

test(`Home's snapshot`, () => {
  const renderer = new ShallowRenderer();
  renderer.render(
      <Home
        places={[
          {
            id: 1,
            src: `img`,
            price: `10`,
            rating: `10`,
            name: `Beautiful`,
            type: `Private`
          }
        ]}
        onCardHeaderClick={jest.fn()}
      />
  );
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
