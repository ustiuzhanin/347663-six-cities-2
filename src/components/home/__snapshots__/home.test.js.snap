// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home's snapshot 1`] = `
<section>
  <h1>
    Home Page
  </h1>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
    >
      <symbol
        id="icon-arrow-select"
        viewBox="0 0 7 4"
      >
        <path
          clipRule="evenodd"
          d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        id="icon-bookmark"
        viewBox="0 0 17 18"
      >
        <path
          d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
        />
      </symbol>
      <symbol
        id="icon-star"
        viewBox="0 0 13 12"
      >
        <path
          clipRule="evenodd"
          d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
          fillRule="evenodd"
        />
      </symbol>
    </svg>
  </div>
  <div
    className="page page--gray page--main"
  >
    <header
      className="header"
    >
      <div
        className="container"
      >
        <div
          className="header__wrapper"
        >
          <div
            className="header__left"
          >
            <a
              className="header__logo-link header__logo-link--active"
            >
              <img
                alt="6 cities logo"
                className="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            className="header__nav"
          >
            <ul
              className="header__nav-list"
            >
              <li
                className="header__nav-item user"
              >
                <a
                  className="header__nav-link header__nav-link--profile"
                  href="#"
                >
                  <div
                    className="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    className="header__user-name user__name"
                  >
                    Oliver.conner@gmail.com
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      className="page__main page__main--index"
    >
      <h1
        className="visually-hidden"
      >
        Cities
      </h1>
      <div
        className="tabs"
      >
        <section
          className="locations container"
        >
          <ul
            className="locations__list tabs__list"
          >
            <Memo(Connect(CityList))
              offers={
                Array [
                  Object {
                    "bedrooms": 1,
                    "city": Object {
                      "location": Object {
                        "latitude": 51.225402,
                        "longitude": 6.776314,
                        "zoom": 13,
                      },
                      "name": "Dusseldorf",
                    },
                    "description": "This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.",
                    "goods": Array [
                      "Breakfast",
                      "Laptop friendly workspace",
                    ],
                    "host": Object {
                      "avatar_url": "img/avatar-angelina.jpg",
                      "id": 25,
                      "is_pro": true,
                      "name": "Angelina",
                    },
                    "id": 1,
                    "images": Array [
                      "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/5.jpg",
                      "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/19.jpg",
                      "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/1.jpg",
                      "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/2.jpg",
                      "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/10.jpg",
                      "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/13.jpg",
                      "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/20.jpg",
                      "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg",
                      "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/4.jpg",
                      "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg",
                      "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/8.jpg",
                      "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg",
                      "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/11.jpg",
                      "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/16.jpg",
                    ],
                    "is_favorite": false,
                    "is_premium": false,
                    "location": Object {
                      "latitude": 51.248402000000006,
                      "longitude": 6.763314,
                      "zoom": 16,
                    },
                    "max_adults": 2,
                    "preview_image": "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg",
                    "price": 101,
                    "rating": 3.1,
                    "title": "Canal View Prinsengracht",
                    "type": "room",
                  },
                ]
              }
            />
          </ul>
        </section>
      </div>
      <div
        className="cities"
      >
        <div
          className="cities__places-container container"
        >
          <Memo(Connect(CardList))
            offers={
              Array [
                Object {
                  "bedrooms": 1,
                  "city": Object {
                    "location": Object {
                      "latitude": 51.225402,
                      "longitude": 6.776314,
                      "zoom": 13,
                    },
                    "name": "Dusseldorf",
                  },
                  "description": "This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.",
                  "goods": Array [
                    "Breakfast",
                    "Laptop friendly workspace",
                  ],
                  "host": Object {
                    "avatar_url": "img/avatar-angelina.jpg",
                    "id": 25,
                    "is_pro": true,
                    "name": "Angelina",
                  },
                  "id": 1,
                  "images": Array [
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/5.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/19.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/1.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/2.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/10.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/13.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/20.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/4.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/8.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/11.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/16.jpg",
                  ],
                  "is_favorite": false,
                  "is_premium": false,
                  "location": Object {
                    "latitude": 51.248402000000006,
                    "longitude": 6.763314,
                    "zoom": 16,
                  },
                  "max_adults": 2,
                  "preview_image": "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg",
                  "price": 101,
                  "rating": 3.1,
                  "title": "Canal View Prinsengracht",
                  "type": "room",
                },
              ]
            }
            onCardHeaderClick={[MockFunction]}
          />
          <div
            className="cities__right-section"
          >
            <section
              className="cities__map map"
            >
              <Memo(Connect(RenderMap)) />
            </section>
          </div>
        </div>
      </div>
    </main>
  </div>
</section>
`;
